addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-service.test.js

```diff
@@ -644,6 +644,111 @@ describe('EnvironmentSCService', () => {
       ]);
       AWSMock.restore();
     });
+
+    it('Should call listNotebookInstances with NextToken if NextToken is returned', async () => {
+      // BUILD
+      indexesService.list = jest
+        .fn()
+        .mockResolvedValue([{ id: 'some-index-id', awsAccountId: 'some-aws-account-uuid' }]);
+      awsAccountsService.list = jest.fn().mockResolvedValue([
+        {
+          roleArn: 'some-role-arn',
+          externalId: 'some-external-id',
+          id: 'some-aws-account-uuid',
+          accountId: 'some-aws-account-id',
+        },
+      ]);
+      dbService.table.scan.mockResolvedValueOnce([
+        {
+          id: 'some-environment-id',
+          indexId: 'some-index-id',
+          status: 'STOPPING',
+          outputs: [{ OutputKey: 'NotebookInstanceName', OutputValue: 'notebook-instance-name' }],
+        },
+        {
+          id: 'some-environment-id-1',
+          indexId: 'some-index-id',
+          status: 'STOPPING',
+          outputs: [{ OutputKey: 'NotebookInstanceName', OutputValue: 'notebook-instance-name-1' }],
+        },
+        {
+          id: 'some-environment-id-2',
+          indexId: 'some-index-id',
+          status: 'STOPPING',
+          outputs: [{ OutputKey: 'NotebookInstanceName', OutputValue: 'notebook-instance-name-2' }],
+        },
+      ]);
+      service.update = jest.fn();
+
+      AWSMock.setSDKInstance(aws.sdk);
+      AWSMock.mock('STS', 'assumeRole', { Credentials: stsResponse });
+      let listNoteBookTimes = 2;
+      AWSMock.mock('SageMaker', 'listNotebookInstances', (params, callback) => {
```

Comment:
Instead of using `listNotebookTimes` to track the order of the calls, I think we can use a `jest.fn` and passing it into `AWSMock`. 

For an example.
```
const testFn = jest.fn();

testFn
    .mockImplementation(() => {
        return true;
    })
    .mockImplementation((params, callback) => {
        return params.shouldPass;
    });

console.log(`Result: ${testFn({ shouldPass: false })}`);
console.log(`Result: ${testFn({ shouldPass: true })}`)
```

Use the correct `callback` and `params` though for dealing with the `NextToken`

Reply:
Neat! Thanks Tim, will update.